src/cluster_legacy.c

```diff
@@ -5663,35 +5671,32 @@ void addNodeDetailsToShardReply(client *c, clusterNode *node) {
     addReplyBulkCString(c, nodeIsSlave(node) ? "replica" : "master");
     reply_count++;
 
-    addReplyBulkCString(c, "replication-offset");
-    addReplyLongLong(c, node_offset);
-    reply_count++;
+    if (!topology) {
+        addReplyBulkCString(c, "replication-offset");
+        addReplyLongLong(c, node_offset);
+        reply_count++;
 
-    addReplyBulkCString(c, "health");
-    const char *health_msg = NULL;
-    if (nodeFailed(node)) {
-        health_msg = "fail";
-    } else if (nodeIsSlave(node) && node_offset == 0) {
-        health_msg = "loading";
-    } else {
-        health_msg = "online";
+        addReplyBulkCString(c, "health");
+        const char *health_msg = NULL;
+        if (nodeFailed(node)) {
+            health_msg = "fail";
+        } else if (nodeIsSlave(node) && node_offset == 0) {
+            health_msg = "loading";
+        } else {
+            health_msg = "online";
+        }
+        addReplyBulkCString(c, health_msg);
+        reply_count++;
     }
-    addReplyBulkCString(c, health_msg);
-    reply_count++;
 
     setDeferredMapLen(c, node_replylen, reply_count);
 }
 
 /* Add the shard reply of a single shard based off the given primary node. */
-void addShardReplyForClusterShards(client *c, list *nodes) {
-    serverAssert(listLength(nodes) > 0);
-    clusterNode *n = listNodeValue(listFirst(nodes));
+void addShardReplyForClusterShards(client *c, clusterNode* n, int topology) {
```

Comment:
minor: `clusterNode* n` -> `clusterNode* primary` to clarify that n is a primary node